SimPEG/VRM/ProblemVRM.py

```diff
@@ -0,0 +1,609 @@
+from SimPEG import Problem, mkvc, Maps, Props, Solver as simpegSolver
+from SimPEG.VRM.SurveyVRM import SurveyVRM
+# from SimPEG.VRM.FieldsVRM import Fields_LinearFWD
+import numpy as np
+import scipy.sparse as sp
+import discretize
+
+############################################
+# BASE VRM PROBLEM CLASS
+############################################
+
+
+class BaseProblemVRM(Problem.BaseProblem):
+    """
+Base class for VRM problem.
+
+REQUIRED ARGUMENTS:
+
+    mesh: 3D tensor or OcTree mesh
+
+KWARGS:
+
+    refFact: Maximum refinement factor for sensitivities (default = 3)
+    refRadius: Distances from source in which cell sensitivities are refined.
+               Must be an array or list with elements equal to the refFact.
+               (default based on minimum cell size)
+    topoMap: A Maps object which operates from mesh cells to cells which are
+             computed in the forward model (default is all mesh cells)
+
+    """
+
+    # SET CLASS ATTRIBUTES
+    _refFact = None
+    _refRadius = None
+    _topoMap = None
+    _AisSet = False
+    _TisSet = False
+    surveyPair = SurveyVRM
+
+    def __init__(self, mesh, **kwargs):
+
+        # **kwargs
+        self._refFact = kwargs.get('refFact', 3)
```

Comment:
refFact refers to refinement factor. The idea is that many cells near the transmitter do not experience uniform magnetization due to the variation in the shape of the primary field. Instead of refining the mesh, we let the physical property of that cell remain constant but we recompute the sensitivity of that cell to the data.

Reply:
An explanation for this starts on pg 48 of my thesis